### ISSUE TITLE
Training problem

### ISSUE THREAD

**Issue by Gitchenguang** [MichalBusta/DeepTextSpotter#53]
Hi, I want to finetune based the models you shared. And
1. Load the shared models' weights:
In train.py line:417 after "nets = create_solvers_tiny(args)", I added "nets[0].net.copy_from('models/tiny.caffemodel')
 nets[1].net.copy_from('models/model.caffemodel')"
2. Prepare the dataset 
For "SynthText in the Wild Dataset"  
One gt example and compute process is below:
0 0.486886901847 0.633320810672 0.0935274765903 0.0859538287988 0.0876311351135 with  
 0 0.797795969632 0.0569849947458 0.149034182834 0.0220692088457 -0.017203329202 Sender:   
 0 0.814664408352 0.116348712709 0.190254920833 0.0202967339849 -0.0218602353083 BNR-RTP  
 0 0.761555379219 0.228247321585 0.124532280215 0.0220434690512 -0.0543050168696 Lines:

![image](https://user-images.githubusercontent.com/5471010/44759763-6c81d380-ab6e-11e8-8e61-554b04495f36.png)
The  icdar2013 test dataset is used and prepared for validation dataset , the vocabulary dictionary is generated from the ground truths in icdar2013 test dataset 
3. Set the datasets path in the train.py and data source in tiny.prototxt. All other parameters are as default

`parser = argparse.ArgumentParser()`
`parser.add_argument('-data_dir', default='/data/PublicDataSets/SyntheticTextInWild')`
`parser.add_argument('--train_list', default='/data/PublicDataSets/SyntheticTextInWild/all_list.txt')`
`parser.add_argument('-valid_list', default='/data/PublicDataSets/3-2013ICDAR-Test-Set/evaluationlist.txt')`
`parser.add_argument('-debug', type=int, default=0)`
`parser.add_argument('-batch_size', type=int, 'default=4')`

4. However, after i run train.py the results is 
90 - lr:1.000e-04 ctc:26.2039/13.2244 wr:0.00/0.00, loc:2.00 28
EaHii --- UPGRADE
VRAn --- POCKET
VEAR --- ASDA
PIGalARD --- ASDA
ATARE --- ASDA
AED --- ASDA
AnGrD --- LCD2470WVX
AG --- MultiSync
AD --- plan
I0829 09:40:07.971902 21025 solver.cpp:218] Iteration 100 (2.11281 iter/s, 47.3303s/100 iters), loss = 34.1541
I0829 09:40:07.972014 21025 sgd_solver.cpp:105] Iteration 100, lr = 9.92565e-05
AD --- plan
UlAD --- V03
UsN --- art
MD --- plan
100 - lr:1.000e-04 ctc:7.9630/17.0779 wr:0.00/0.00, loc:1.12 17
I0829 09:40:08.735069 21025 net.cpp:676] Ignoring source layer caffe.Dropout_35
  E2E recall 0.000 / 0.500
  E2E recall 0.000 / 2.000
  E2E recall 0.000 / 0.750
  E2E recall 0.000 / 1.000
  E2E recall 0.000 / 1.000
  E2E recall 0.000 / 1.000
  E2E recall 0.000 / 1.000
  E2E recall 0.000 / 1.000
  E2E recall 0.000 / 0.500
  E2E recall 0.000 / 1.000
  E2E recall 0.000 / 1.125
  E2E recall 0.000 / 0.500
  E2E recall 0.000 / 1.000
  E2E recall 0.000 / 0.667
  E2E recall 0.000 / 0.000
  E2E recall 0.000 / 1.000
  E2E recall 0.000 / 0.714
  E2E recall 0.000 / 0.000
  E2E recall 0.000 / 1.000
  E2E recall 0.000 / 1.250
  E2E recall 0.000 / 0.800
  E2E recall 0.000 / 1.500
E2E recall 0.000 / 0.821

Could you please give me some ideas on my problem? Thanks!

**Comment by Linchenguang** [comment]
The image sizes are kept unchanged and differ with each other, Should I resize them into the same size?

**Comment by Gitchenguang** [comment]
Pre-train is important ,and .solverstate files are needed for end-to-end training

**Comment by mowkee** [comment]
@Linchenguang Have you been able to train since you have closed the issue? could you please share your knowledge?

**Comment by Gitchenguang** [comment]
I am able to train the network, but the performance of models i trained were not very good so far.

**Comment by mowkee** [comment]
ok could you tell me how did you train the network? how did you prepare the data to train the network? did you use `train.py` to train the network? it requires some text files and a data set but I don't find the txt files in the data set I have downloaded. I would really appreciate if you could write a guide about it.

**Comment by Gitchenguang** [comment]
First, I wrote some python files to generate the train data and corresponding txt files. then, I modified the train.py to pretrain the  RPN network and CTC network separately, and obtained .solverstate and .caffemodel files, these files are used to restore  the corresponding networks when you run train.py to train the networks. Finally, I just run train.py and it works.

**Comment by Gitchenguang** [comment]
By the way, the figure above to compute the angle is wrong, the sign is opposite

**Comment by mowkee** [comment]
would you care to share your code for training please? I would really appreciate your help because I've not been able to understand how to do it and I've been struggling for 2 months.

**Comment by Gitchenguang** [comment]
Ok,, if you need, i will share them one of these days，but I do not know if am i right

**Comment by mowkee** [comment]
thank you so much. I don't know how to thank you. I really appreciate your prompt response and help.

**Comment by Gitchenguang** [comment]
Hello, do you have any emails? I send the detection pretrain .py file and ctc pretrain .py file to you, or you can send an email to me chen@zchenguang.com

**Comment by DachengLiu1994** [comment]
hi, @Linchenguang  , Can you send the detection pretrain .py file and ctc pretrain .py file to me,my mail address is dacheng1235@163.com.
Thank you very much!

**Comment by nabeel3133** [comment]
hi @Linchenguang, I have a few images around 60 that I want to train on Busta. Can you guide me on how to train the images?
